### ISSUE TITLE
Resnet pretrained with Noisy student

### ISSUE THREAD

**Issue by JulienMaille** [qubvel/segmentation_models#329]
I've been playing with timm-efficiennet and noticed I'm converging with a better IoU when loading the model pretrained with noisy-student. Made me wonder if there are some similar pre-trained resnet18 available?

**Comment by JordanMakesMaps** [comment]
Probably not for ResNet-18, it's a pretty small and old architecture compared to the EfficientNet series, and the dataset that they used with Noisy-student is a bit more modern.

But, you might some other pre-trained weights [here](https://www.kaggle.com/c/planet-understanding-the-amazon-from-space/discussion/31862)

**Comment by JulienMaille** [comment]
Small and old but still relevant!
BTW what could explain that I converge to a better score with noisy-student? Shouldn't I just converge faster?

**Comment by JordanMakesMaps** [comment]
Definitely still relevant, just depends on the size of the dataset :)

It might be because the model that produced the Noisy-Student weights did better than the same model that produced the ImageNet weights on the ImageNet dataset.

![image](https://user-images.githubusercontent.com/8714759/81882815-a58bf280-9561-11ea-8932-94520f513299.png)

**Comment by JulienMaille** [comment]
I can get an IoU of 0.630 with timm-efficientnet-b4 + noisy-student but only 0.610 with ImageNet.
However I prefer the speed of resnet18, that's why I wished I could get the same improvement!

**Comment by JordanMakesMaps** [comment]
"Is is realistic/possible to train it myself"

"Noisy Student Training extends the idea of self-training and distillation with the use of equal-or-larger student models and noise added to the student during learning. On ImageNet, we _first_ train an EfficientNet model on labeled images and use it as a teacher to generate pseudo labels for **300M unlabeled images**. We _then_ train a _larger_ EfficientNet as a student model on the combination of labeled and pseudo labeled images. We **iterate** this process by putting back the student as the teacher. " 

[Paper](https://arxiv.org/abs/1911.04252)

You can try, but this achievement was made by some researchers at Google that access to a godly amount computation resources :)

What is timm?

**Comment by JulienMaille** [comment]
My bad, I posted by mistake in the tf version of the library. timm efficientnet encoders were recently added to the pytorch version of this lib.
https://github.com/qubvel/segmentation_models.pytorch/pull/189
